tests/unit-tests/test_construct_graph.py

```diff
@@ -116,7 +117,102 @@ def test_label():
+        assert id1 == int(id2)
+
+    # Test the case that the ID array of integer type
+    try:
+        rand_ids = np.random.randint(10, size=5)
+        remap_ids, idx = id_map(rand_ids)
+        raise ValueError("fails")
+    except:
+        pass
+
+    # Test the case that the ID map has integer keys.
+    str_ids = np.array([i for i in range(10)])
+    id_map = IdMap(str_ids)
+    try:
+        rand_ids = np.array([str(random.randint(0, len(str_ids))) for _ in range(5)])
+        remap_ids, idx = id_map(rand_ids)
+        raise ValueError("fails")
+    except:
+        pass
+
+def test_map_node_ids():
+    from graphstorm.gconstruct.construct_graph import IdMap
+    from graphstorm.gconstruct.construct_graph import map_node_ids
+    str_src_ids = np.array([str(i) for i in range(10)])
+    str_dst_ids = np.array([str(i) for i in range(15)])
+    id_map = {"src": IdMap(str_src_ids),
+              "dst": IdMap(str_dst_ids)}
+
+    # Test the case that both source node IDs and destination node IDs exist.
+    src_ids = np.array([str(random.randint(0, len(str_src_ids) - 1)) for _ in range(15)])
+    dst_ids = np.array([str(random.randint(0, len(str_dst_ids) - 1)) for _ in range(15)])
+    new_src_ids, new_dst_ids = map_node_ids(src_ids, dst_ids, ("src", None, "dst"),
+                                            id_map, False)
+    assert len(new_src_ids) == len(src_ids)
+    assert len(new_dst_ids) == len(dst_ids)
+    for src_id1, src_id2 in zip(new_src_ids, src_ids):
+        assert src_id1 == int(src_id2)
+    for dst_id1, dst_id2 in zip(new_dst_ids, dst_ids):
+        assert dst_id1 == int(dst_id2)
+
+    # Test the case that source node IDs don't exist.
+    src_ids = np.array([str(random.randint(0, 20)) for _ in range(15)])
+    dst_ids = np.array([str(random.randint(0, len(str_dst_ids) - 1)) for _ in range(15)])
+    try:
+        new_src_ids, new_dst_ids = map_node_ids(src_ids, dst_ids, ("src", None, "dst"),
+                                                id_map, False)
+        raise ValueError("fail")
+    except:
+        pass
+    # Test the case that source node IDs don't exist and we skip non exist edges.
+    new_src_ids, new_dst_ids = map_node_ids(src_ids, dst_ids, ("src", None, "dst"),
+                                            id_map, True)
+    num_valid = sum([int(id_) < len(str_src_ids) for id_ in src_ids])
+    assert len(new_src_ids) == num_valid
+    assert len(new_dst_ids) == num_valid
+
+    # Test the case that destination node IDs don't exist.
+    src_ids = np.array([str(random.randint(0, len(str_src_ids) - 1)) for _ in range(15)])
+    dst_ids = np.array([str(random.randint(0, 20)) for _ in range(15)])
+    try:
+        new_src_ids, new_dst_ids = map_node_ids(src_ids, dst_ids, ("src", None, "dst"),
+                                                id_map, False)
+        raise ValueError("fail")
+    except:
+        pass
+    # Test the case that destination node IDs don't exist and we skip non exist edges.
```

Comment:
Shall we split it into multiple small unitests?

Reply:
do you mean split it into multiple `check_xxx` functions?